### ISSUE TITLE
No emails are being sent.

### ISSUE THREAD

**Issue by stevenmcastano** [Mailtrain-org/mailtrain#106]
I've got a list, template, campaign, and when I enter my SMTP settings everything verifies and says I'm ready to send mail.

However, when I click to start sending a campaign, I see the spinning arrows saying it's sending, but it never makes and progress forward and looking at the logs on me email server, it's not even attempting to send the emails. Not a single connection.

I have the box checked to log SMTP transactions also, but I can't seem to find the log or anything that might be wrong with it.

What steps should I take next to troubleshoot?

**Comment by andris9** [comment]
Do the subscription confirmation emails work?

**Comment by stevenmcastano** [comment]
Yes, they do and based on the email headers, they're being sent through the proper server via SSL, all my DKIM and SPF records are in place, everything looks good to go. I tried subscribing with emails from gmail, yahoo, and two of my own custom domains.

**Comment by stevenmcastano** [comment]
Out of nowhere a test campaign I left trying to send HOURS ago to a list with just my email address on it just went through without even doing anything... it's just been sitting there saying it was sending without doing anything for hours.

Any ideas?

**Comment by stevenmcastano** [comment]
Now the other campaign to the original 30 or so users just all of a sudden went out, and the odd part is that I set it to sent right after the last single email went out, which is when I posted here last... which was 5 hours ago and looking at the list of delivered emails... they all say they were sent/delivered 5 hours ago as well in mailtrain, but the log on the server as well as two of my own email addresses on the list all show the emails were just sent a few minutes ago. So it seems like they're on some sort of almost exact 5 hour delay.

Is there someone with time zones? delays? other time configuration? etc....?

**Comment by stevenmcastano** [comment]
Even users clicks that are happening right now as they're getting the emails are still being recorded as having happened "5 hours ago"

**Comment by andris9** [comment]
Yes, seems like a timezone issue. Could you try modifying the mysql config and change timezone from "local" to "-0500" or "0500" to see if the shift disappears?

**Comment by stevenmcastano** [comment]
Looks like the timezone remove the shift in times and it looks like it got emails to send ASAP... I only tried it to my test list of one users, but this time when it hit send it prepped the email and sent right away. I'll have to give it some time until I have another excuse to both my customers.

Any idea why the timezone setting would actually delay email sending??

**Comment by andris9** [comment]
When you start a campaign, then you can also set the delay time. This time (hours and minutes) are added to current timestamp and this timestamp is stored to MySQL as the scheduled start time for the campaign. In the background the sending job periodically checks if there are any campaigns that have scheduled start time older than NOW().

**Comment by stevenmcastano** [comment]
That makes sense.... it's something that really should be in the documentation somewhere I'd say though, that seems pretty important.

**Comment by santimacia** [comment]
Can you give more details on this? I have the same issue.

**Comment by vikash32** [comment]
I am also running with the same issue, @stevenmcastano Please let us know how you resolved it. Where i need to update the timezone

**Comment by rnarayan** [comment]
I am also facing the same problem. No emails are being sent from campaign.

I installed MailTrain on at Linode on Ubuntu 16.04. Right out of the box, I was able to register a use using subscription form. An email was sent (I think it was using ZONE_MTA). I clicked the "Subscribe" button in the email to confirm subscription. I repeated it for one more email. Everything perfect.

Then I created a Campaign for these two test users. When I want to send the campaign, I get an endless spinning wheel. At this point, it's still MailTrain out of the box. ZONE_MTA is not sending the campaign.

As a second try, I configured MailTrain to use Amazon SES API. (In tests at Amazon, SES works and is properly configured.) When I click on "Check Mailer Config" on MailTrain settings, it gives a "Success" message - i.e. it was able to connect with Amazon SES.

Then I created a new campaign but the result was the same. Endless spinning wheel. Campaign is not sent out.

Server and MariaDB timezone is the same. So no problem there.

At this time, I have not been able to send out a single campaign, whether using ZONE_MTA or Amazon SES API.

Would love to get a solution for this.

**Comment by addelindh** [comment]
Check if you've got verp enabled in the /opt/mailtrain/config/production.toml file

[verp]
enabled=true

Set it to false and restart the application or server, seems that if verp is enabled but not configured, campaign emails are not sent.

**Comment by w3update** [comment]
i have configured VERP bounce handling but facing the same issue even set current country timezone with ntp sync, even mysql timezone return exact system one as mysql> SELECT @@global.time_zone, @@session.time_zone;

**Comment by w3update** [comment]
I guess the issue related to VERP only as i disabled and restart it worked.

will try to fix it.

thanks

**Comment by christiangenco** [comment]
My emails also aren't getting sent.

I've been able to send them before, and my "Check Mailer config" button on `/settings` returns `Mailer settings verified, ready to send some mail!`. 

I just did a `cd /opt/mailtrain; git pull` and am on [ec7699](https://github.com/Mailtrain-org/mailtrain/commit/ec76994f694e73d4217898d405d4f236830eb6a4). After upgrading I did `npm install --production; service mailtrain restart`.

My `/opt/mailtrain/config/production.toml` looks like this:

    user="mailtrain"
    group="mailtrain"
    [log]
    level="verbose"
    [www]
    port=80
    secret="MYSECRET"
    [mysql]
    password="MYPASSWORD"
    timezone="-0600"
    [redis]
    enabled=true

(I updated `timezone="-0600"` per [@andris9's recommendation](https://github.com/Mailtrain-org/mailtrain/issues/106#issuecomment-261443119)).

Strangely, I can't find a mailtrain logfile. It's not at `/var/log/mailtrain.log`, and even `find / -name "mailtrain.log"` finds nothing.

I also tried setting the scheduled time to `-12`, with the same result:

<img width="739" alt="screen shot 2017-04-27 at 12 33 47 pm" src="https://cloud.githubusercontent.com/assets/179361/25496261/da544fa0-2b45-11e7-9eec-bb2fbac63b9a.png">

What else should I try?

**Comment by witzig** [comment]
> Strangely, I can't find a mailtrain logfile. It's not at /var/log/mailtrain.log, and even find / -name "mailtrain.log" finds nothing.

Does `journalctl -f -u mailtrain` yield something?

**Comment by christiangenco** [comment]
@witzig Ooo, yes. Nothing suspicious shows up there when going through the campaign sending flow:

    Apr 27 14:15:19 mailtrain node[17756]: info HTTP GET /bootstrap/js/bootstrap.min.js 304 1.038 ms - -
    Apr 27 14:15:19 mailtrain node[17756]: info HTTP GET /javascript/tables.js 304 1.320 ms - -
    Apr 27 14:15:19 mailtrain node[17756]: info HTTP GET /moment/moment.min.js 304 0.230 ms - -
    Apr 27 14:15:19 mailtrain node[17756]: info HTTP GET /bootstrap/fonts/glyphicons-halflings-regular.woff2 304 0.594 ms - -
    Apr 27 14:15:19 mailtrain node[17756]: info HTTP GET /javascript/fetch.js 304 2.242 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP POST /campaigns/send 302 24.999 ms - 78
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /campaigns/view/1 200 6.655 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /datatables/datatables.min.css 304 0.886 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /bootstrap/themes/united.min.css 304 1.247 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /css/footer.css 304 0.807 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /datepicker/css/bootstrap-datepicker3.css 304 1.046 ms - -
    Apr 27 14:15:28 mailtrain node[17756]: info HTTP GET /css/mailtrain.css 304 1.319 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /datepicker/js/bootstrap-datepicker.min.js 304 1.076 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /bootstrap/js/bootstrap.min.js 304 2.447 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /javascript/jquery-2.2.1.min.js 304 2.755 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /datatables/datatables.min.js 304 3.059 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /javascript/tables.js 304 0.467 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /moment/moment.min.js 304 0.750 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /bootstrap/fonts/glyphicons-halflings-regular.woff2 304 0.358 ms - -
    Apr 27 14:15:29 mailtrain node[17756]: info HTTP GET /javascript/fetch.js 304 0.218 ms - -

**Comment by christiangenco** [comment]
I paused a campaign yesterday, and when resumed it sent right away. Sounds like it might still be a time zone error?

**Comment by christiangenco** [comment]
Yup - it was. I think I fixed it by restarting mysql:

    /etc/init.d/mysql restart

Before I restarted mysql, when it was 14:19 on my wall clock in the central time zone, I got:

    $ mysql
    MariaDB [(none)]> use mailtrain
    MariaDB [mailtrain]> select now();
    +---------------------+
    | now()               |
    +---------------------+
    | 2017-04-28 12:18:25 |
    +---------------------+
    1 row in set (0.00 sec)

ie: the mysql clock was two hours behind. Presumably my campaigns would send two afters after I hit "send" in this configuration.

After restarting mysql with `/etc/init.d/mysql restart`, I got:

    MariaDB [mailtrain]> select now();
    +---------------------+
    | now()               |
    +---------------------+
    | 2017-04-28 14:19:13 |
    +---------------------+
    1 row in set (0.00 sec)

ie: the correct hour. Strangely, a campaign I had tried to send before changing the time zone now said "Sending scheduled in an hour" (I had queued it an hour ago). Stopping it and re-sending sent it immediately.

tl;dr: **to fix mail not sending**:


0. Confirm it's a time zone issue by checking that `$ mysql -e "select now();"` and `$ date` are different.
1. Upgrade mailtrain with `cd /opt/mailtrain; git pull; npm install --production; service mailtrain restart`
2. Make sure there's nothing weird in `/opt/mailtrain/config/production.toml` for the `[mysql]` `timezone`. You should be able to delete the `timezone` line and have it correctly default to `timezone="local"` in `/opt/mailtrain/config/default.toml`
3. Restart mysql with `/etc/init.d/mysql restart`
4. Confirm the time from `$ mysql -e "select now();"` matches the time from `$ date`
